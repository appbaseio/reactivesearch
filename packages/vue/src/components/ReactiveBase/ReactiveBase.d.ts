import * as types from '../../types.ts';

export interface ReactiveBaseProps {
	app?: string;
	children?: types.children;
	credentials?: string;
	as?: string;
	headers?: types.headers;
	queryParams?: string;
	theme?: types.style;
	themePreset?: types.themePreset;
	type?: string;
	url?: string;
	mapKey?: string;
	style?: types.style;
	className?: string;
	graphQLUrl?: string;
	transformResponse?: (...args: any[]) => any;
	transformRequest?: (...args: any[]) => any;
	getSearchParams?: () => string;
	setSearchParams?: (newURL: string) => void;
	initialQueriesSyncTime?: number;
	analyticsConfig?: types.analyticsConfig;
	reactivesearchAPIConfig?: types.appbaseConfig;
	initialState?: types.children;
	analytics?: boolean;
	endpoint?: types.endpointConfig;
	contextCollector?: any;
}

declare function ReactiveBase(props: ReactiveBaseProps): JSX.Element;

export default ReactiveBase;
