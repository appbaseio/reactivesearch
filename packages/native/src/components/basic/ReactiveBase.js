import React, { Component } from 'react';
import { createProvider } from 'react-redux';
import { Container } from 'native-base';
import Appbase from 'appbase-js';

import configureStore, { storeKey } from '@appbaseio/reactivecore';

import types from '@appbaseio/reactivecore/lib/utils/types';

import ThemeProvider from '../../theme/ThemeProvider';
import theme from '../../theme';

/* use a custom store key so reactivesearch does not interfere
   with a different redux store in a nested context */
const Provider = createProvider(storeKey);

// for network debugging while development
// GLOBAL.XMLHttpRequest = GLOBAL.originalXMLHttpRequest || GLOBAL.XMLHttpRequest;

class ReactiveBase extends Component {
	constructor(props) {
		super(props);

		this.type = props.type ? props.type : '*';

		const credentials
			= props.url && props.url.trim() !== '' && !props.credentials ? null : props.credentials;

		const config = {
			url: props.url && props.url.trim() !== '' ? props.url : 'https://scalr.api.appbase.io',
			app: props.app,
			credentials,
			type: this.type,
		};

		const { headers = {} } = props;
		const appbaseRef = Appbase(config);
		this.store = configureStore({ config, appbaseRef, headers });
	}

	render() {
		return (
			<ThemeProvider theming={{ ...theme, ...this.props.theme }}>
				<Provider store={this.store}>
					<Container>{this.props.children}</Container>
				</Provider>
			</ThemeProvider>
		);
	}
}

ReactiveBase.propTypes = {
	type: types.string,
	url: types.string,
	credentials: types.string,
	headers: types.headers,
	app: types.stringRequired,
	theme: types.style,
	children: types.children,
};

export default ReactiveBase;
