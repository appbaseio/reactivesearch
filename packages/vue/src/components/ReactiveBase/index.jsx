import configureStore from '@appbaseio/reactivecore';
import VueTypes from 'vue-types';
import Appbase from 'appbase-js';
import Provider from '../Provider';
import { composeThemeObject } from '../../utils/index';
import types from '../../utils/vueTypes';
import URLParamsProvider from '../URLParamsProvider.jsx';
import getTheme from '../../styles/theme';

const URLSearchParams = require('url-search-params');

const ReactiveBase = {
	name: 'ReactiveBase',
	data() {
		this.state = {
			key: '__REACTIVE_BASE__',
		};
		return this.state;
	},
	created() {
		this.setStore(this.$props);
	},
	props: {
		app: types.stringRequired,
		analytics: VueTypes.bool.def(false),
		credentials: types.string,
		headers: types.headers,
		queryParams: types.string,
		theme: VueTypes.object.def({}),
		themePreset: VueTypes.string.def('light'),
		type: types.string,
		url: types.string,
		mapKey: types.string,
		className: types.string,
		initialState: VueTypes.object.def({}),
		transformRequest: types.func,
	},
	provide() {
		return {
			theme: composeThemeObject(getTheme(this.$props.themePreset), this.$props.theme),
			store: this.store,
		};
	},
	watch: {
		app() {
			this.updateState(this.$props);
		},
		url() {
			this.updateState(this.$props);
		},
		type() {
			this.updateState(this.$props);
		},
		credentials() {
			this.updateState(this.$props);
		},
		mapKey() {
			this.updateState(this.$props);
		},
		headers() {
			this.updateState(this.$props);
		},
	},
	methods: {
		updateState(props) {
			this.setStore(props);
			this.setState(state => ({
				key: `${state.key}-0`,
			}));
		},
		setStore(props) {
			const credentials
				= props.url && props.url.trim() !== '' && !props.credentials
					? null
					: props.credentials;
			const config = {
				url:
					props.url && props.url.trim() !== ''
						? props.url
						: 'https://scalr.api.appbase.io',
				app: props.app,
				credentials,
				type: props.type ? props.type : '*',
				transformRequest: props.transformRequest,
				analytics: props.analytics,
			};
			let queryParams = '';

			if (typeof window !== 'undefined') {
				queryParams = window.location.search;
			} else {
				queryParams = props.queryParams || '';
			}

			const params = new URLSearchParams(queryParams);
			let selectedValues = {};

			Array.from(params.keys()).forEach(key => {
				try {
					const parsedParams = JSON.parse(params.get(key));
					const selectedValue = {};
					if (parsedParams.value) {
						selectedValue.value = parsedParams.value;
					} else {
						selectedValue.value = parsedParams;
					}
					if (parsedParams.category) selectedValue.category = parsedParams.category;
					selectedValues = {
						...selectedValues,
						[key]: selectedValue,
					};
				} catch (e) {
					// Do not add to selectedValues if JSON parsing fails.
				}
			});

			const { headers = {}, themePreset } = props;
			const appbaseRef = Appbase(config);

			if (this.$props.transformRequest) {
				appbaseRef.transformRequest = this.$props.transformRequest;
			}

			const initialState = {
				config: {
					...config,
					mapKey: props.mapKey,
					themePreset,
				},
				appbaseRef,
				selectedValues,
				headers,
				...this.$props.initialState,
			};
			this.store = configureStore(initialState);
		},
	},
	render() {
		const children = this.$slots.default;
		const { headers, style, className } = this.$props;
		return (
			<Provider store={this.store}>
				<URLParamsProvider headers={headers} style={style} className={className}>
					{children}
				</URLParamsProvider>
			</Provider>
		);
	},
};
ReactiveBase.install = function(Vue) {
	Vue.component(ReactiveBase.name, ReactiveBase);
};

export default ReactiveBase;
