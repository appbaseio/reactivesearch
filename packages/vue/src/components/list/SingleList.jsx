import { Actions, helper } from '@appbaseio/reactivecore';
import { componentTypes } from '@appbaseio/reactivecore/lib/utils/constants';
import VueTypes from 'vue-types';
import Title from '../../styles/Title';
import Input from '../../styles/Input';
import Container from '../../styles/Container';
import {
	connect,
	getComponent,
	hasCustomRenderer,
	isEvent,
	isFunction,
	getValidPropsKeys,
	updateCustomQuery,
	updateDefaultQuery,
	isQueryIdentical,
} from '../../utils/index';
import types from '../../utils/vueTypes';
import { UL, Radio } from '../../styles/FormControlList';
import { getAggsQuery } from './utils';

const {
	addComponent,
	removeComponent,
	watchComponent,
	updateQuery,
	setQueryOptions,
	setQueryListener,
	updateComponentProps,
	setComponentProps,
	setCustomQuery,
	setDefaultQuery,
} = Actions;
const {
	getQueryOptions,
	pushToAndClause,
	checkValueChange,
	getClassName,
	getOptionsFromQuery,
	isEqual,
	checkSomePropChange,
} = helper;

const SingleList = {
	name: 'SingleList',
	props: {
		beforeValueChange: types.func,
		className: types.string.def(''),
		componentId: types.stringRequired,
		customQuery: types.func,
		dataField: types.stringRequired,
		defaultValue: types.string,
		value: types.value,
		defaultQuery: types.func,
		filterLabel: types.string,
		innerClass: types.style,
		placeholder: VueTypes.string.def('Search'),
		react: types.react,
		render: types.func,
		renderItem: types.func,
		transformData: types.func,
		selectAllLabel: types.string,
		showCount: VueTypes.bool.def(true),
		showFilter: VueTypes.bool.def(true),
		showRadio: VueTypes.bool.def(true),
		showSearch: VueTypes.bool.def(true),
		size: VueTypes.number.def(100),
		sortBy: VueTypes.oneOf(['asc', 'desc', 'count']).def('count'),
		title: types.title,
		URLParams: VueTypes.bool.def(false),
		showMissing: VueTypes.bool.def(false),
		missingLabel: VueTypes.string.def('N/A'),
		nestedField: types.string,
	},
	data() {
		const props = this.$props;
		this.__state = {
			currentValue: '',
			modifiedOptions:
				props.options && props.options[props.dataField]
					? props.options[props.dataField].buckets
					: [],
			searchTerm: '',
		};
		this.internalComponent = `${props.componentId}__internal`;
		return this.__state;
	},
	created() {
		const onQueryChange = (...args) => {
			this.$emit('queryChange', ...args);
		};
		this.setQueryListener(this.$props.componentId, onQueryChange, e => {
			this.$emit('error', e);
		});
		// Update props in store
		this.setComponentProps(this.componentId, this.$props, componentTypes.singleList);
		this.setComponentProps(this.internalComponent, this.$props, componentTypes.singleList);
		// Set custom and default queries in store
		updateCustomQuery(this.componentId, this.setCustomQuery, this.$props, this.currentValue);
		updateDefaultQuery(this.componentId, this.setDefaultQuery, this.$props, this.currentValue);
	},
	mounted() {
		this.setReact(this.$props);
	},
	beforeMount() {
		this.addComponent(this.internalComponent);
		this.addComponent(this.$props.componentId);
		this.updateQueryHandlerOptions(this.$props);

		if (this.selectedValue) {
			this.setValue(this.selectedValue);
		} else if (this.$props.value) {
			this.setValue(this.$props.value);
		} else if (this.$props.defaultValue) {
			this.setValue(this.$props.defaultValue);
		}
	},

	beforeDestroy() {
		this.removeComponent(this.$props.componentId);
		this.removeComponent(this.internalComponent);
	},
	watch: {
		$props: {
			deep: true,
			handler(newVal) {
				const propsKeys = getValidPropsKeys(newVal);
				checkSomePropChange(newVal, this.componentProps, propsKeys, () => {
					this.updateComponentProps(this.componentId, newVal, componentTypes.singleList);
					this.updateComponentProps(
						this.internalComponent,
						newVal,
						componentTypes.singleList,
					);
				});
			},
		},
		react() {
			this.setReact(this.$props);
		},
		options(newVal) {
			this.modifiedOptions = newVal[this.$props.dataField]
				? newVal[this.$props.dataField].buckets
				: [];
		},
		size() {
			this.updateQueryHandlerOptions(this.$props);
		},
		sortBy() {
			this.updateQueryHandlerOptions(this.$props);
		},
		dataField() {
			this.updateQueryHandlerOptions(this.$props);
			this.updateQueryHandler(this.$data.currentValue, this.$props);
		},
		defaultValue(newVal) {
			this.setValue(newVal);
		},
		value(newVal, oldVal) {
			if (!isEqual(newVal, oldVal)) {
				this.setValue(newVal);
			}
		},
		selectedValue(newVal) {
			if (this.$data.currentValue !== newVal) {
				this.setValue(newVal || '');
			}
		},
		defaultQuery(newVal, oldVal) {
			if (!isQueryIdentical(newVal, oldVal, this.$data.currentValue, this.$props)) {
				this.updateDefaultQueryHandler(this.$data.currentValue, this.$props);
			}
		},
		customQuery(newVal, oldVal) {
			if (!isQueryIdentical(newVal, oldVal, this.$data.currentValue, this.$props)) {
				this.updateQueryHandler(this.componentId, this.$data.currentValue, this.$props);
			}
		},
	},
	render() {
		const { selectAllLabel, renderItem, renderError } = this.$props;
		const renderItemCalc = this.$scopedSlots.renderItem || renderItem;
		const renderErrorCalc = this.$scopedSlots.renderError || renderError;

		if (renderErrorCalc && this.error) {
			return isFunction(renderErrorCalc) ? renderErrorCalc(this.error) : renderErrorCalc;
		}
		if (!this.hasCustomRenderer && this.modifiedOptions.length === 0) {
			return null;
		}

		let itemsToRender = this.$data.modifiedOptions;

		if (this.$props.transformData) {
			itemsToRender = this.$props.transformData(itemsToRender);
		}

		return (
			<Container class={this.$props.className}>
				{this.$props.title && (
					<Title class={getClassName(this.$props.innerClass, 'title') || ''}>
						{this.$props.title}
					</Title>
				)}
				{this.renderSearch()}
				{this.hasCustomRenderer ? (
					this.getComponent()
				) : (
					<UL class={getClassName(this.$props.innerClass, 'list') || ''}>
						{selectAllLabel ? (
							<li
								key={selectAllLabel}
								class={`${
									this.$data.currentValue === selectAllLabel ? 'active' : ''
								}`}
							>
								<Radio
									class={getClassName(this.$props.innerClass, 'radio')}
									id={`${this.$props.componentId}-${selectAllLabel}`}
									name={this.$props.componentId}
									value={selectAllLabel}
									onClick={this.handleClick}
									readOnly
									show={this.$props.showRadio}
									{...{
										domProps: {
											checked: this.$data.currentValue === selectAllLabel,
										},
									}}
								/>
								<label
									class={getClassName(this.$props.innerClass, 'label') || null}
									for={`${this.$props.componentId}-${selectAllLabel}`}
								>
									{selectAllLabel}
								</label>
							</li>
						) : null}
						{itemsToRender
							.filter(item => {
								if (String(item.key).length) {
									if (this.$props.showSearch && this.$data.searchTerm) {
										return String(item.key)
											.toLowerCase()
											.includes(this.$data.searchTerm.toLowerCase());
									}

									return true;
								}

								return false;
							})
							.map(item => (
								<li
									key={item.key}
									class={`${
										this.currentValue === String(item.key) ? 'active' : ''
									}`}
								>
									<Radio
										class={getClassName(this.$props.innerClass, 'radio')}
										id={`${this.$props.componentId}-${item.key}`}
										name={this.$props.componentId}
										value={item.key}
										readOnly
										onClick={this.handleClick}
										type="radio"
										show={this.$props.showRadio}
										{...{
											domProps: {
												checked: this.currentValue === String(item.key),
											},
										}}
									/>
									<label
										class={
											getClassName(this.$props.innerClass, 'label') || null
										}
										for={`${this.$props.componentId}-${item.key}`}
									>
										{renderItemCalc ? (
											renderItemCalc({
												label: item.key,
												count: item.doc_count,
												isChecked: this.currentValue === String(item.key),
											})
										) : (
											<span>
												{item.key}
												{this.$props.showCount && (
													<span
														class={
															getClassName(
																this.$props.innerClass,
																'count',
															) || null
														}
													>
														&nbsp;(
														{item.doc_count})
													</span>
												)}
											</span>
										)}
									</label>
								</li>
							))}
					</UL>
				)}
			</Container>
		);
	},

	methods: {
		setReact(props) {
			const { react } = props;

			if (react) {
				const newReact = pushToAndClause(react, this.internalComponent);
				this.watchComponent(props.componentId, newReact);
			} else {
				this.watchComponent(props.componentId, {
					and: this.internalComponent,
				});
			}
		},

		setValue(nextValue, props = this.$props) {
			let value = nextValue;

			if (nextValue === this.$data.currentValue) {
				value = '';
			}

			const performUpdate = () => {
				this.currentValue = value;
				this.updateQueryHandler(value, props);
				this.$emit('valueChange', value);
			};

			checkValueChange(props.componentId, value, props.beforeValueChange, performUpdate);
		},

		updateDefaultQueryHandler(value, props) {
			let defaultQueryOptions;
			let query = SingleList.defaultQuery(value, props);
			if (this.defaultQuery) {
				const defaultQueryToBeSet = this.defaultQuery(value, props) || {};
				if (defaultQueryToBeSet.query) {
					({ query } = defaultQueryToBeSet);
				}
				defaultQueryOptions = getOptionsFromQuery(defaultQueryToBeSet);
				// Update calculated default query in store
				updateDefaultQuery(props.componentId, this.setDefaultQuery, props, value);
			}
			this.setQueryOptions(this.internalComponent, defaultQueryOptions);
			this.updateQuery({
				componentId: this.internalComponent,
				query,
				value,
				componentType: componentTypes.singleList,
			});
		},

		updateQueryHandler(value, props) {
			const { customQuery } = props;
			let query = SingleList.defaultQuery(value, props);
			let customQueryOptions;
			if (customQuery) {
				({ query } = customQuery(value, props) || {});
				customQueryOptions = getOptionsFromQuery(customQuery(value, props));
				updateCustomQuery(props.componentId, this.setCustomQuery, props, value);
			}
			this.setQueryOptions(props.componentId, customQueryOptions);
			this.updateQuery({
				componentId: props.componentId,
				query,
				value,
				label: props.filterLabel,
				showFilter: props.showFilter,
				URLParams: props.URLParams,
				componentType: componentTypes.singleList,
			});
		},

		generateQueryOptions(props) {
			const queryOptions = getQueryOptions(props);
			return getAggsQuery(queryOptions, props);
		},

		updateQueryHandlerOptions(props) {
			const queryOptions = SingleList.generateQueryOptions(props);
			if (props.defaultQuery) {
				const value = this.$data.currentValue;
				const defaultQueryOptions = getOptionsFromQuery(props.defaultQuery(value, props));
				this.setQueryOptions(this.internalComponent, {
					...queryOptions,
					...defaultQueryOptions,
				});
			} else {
				this.setQueryOptions(this.internalComponent, queryOptions);
			}
		},

		handleInputChange(e) {
			const { value } = e.target;
			this.searchTerm = value;
		},

		renderSearch() {
			if (this.$props.showSearch) {
				return (
					<Input
						class={getClassName(this.$props.innerClass, 'input') || ''}
						onInput={this.handleInputChange}
						value={this.$data.searchTerm}
						placeholder={this.$props.placeholder}
						style={{
							margin: '0 0 8px',
						}}
						themePreset={this.$props.themePreset}
					/>
				);
			}

			return null;
		},

		getComponent() {
			const { currentValue, modifiedOptions } = this.$data;
			const { transformData } = this.$props;
			let itemsToRender = modifiedOptions;
			if (transformData) {
				itemsToRender = transformData(itemsToRender);
			}
			const data = {
				error: this.error,
				loading: this.isLoading,
				value: currentValue,
				data: itemsToRender,
				rawData: this.rawData,
				handleChange: this.handleClick,
			};
			return getComponent(data, this);
		},

		handleClick(e) {
			let currentValue = e;
			if (isEvent(e)) {
				currentValue = e.target.value;
			}
			const { value } = this.$props;
			if (value === undefined) {
				this.setValue(currentValue);
			} else {
				this.$emit('change', currentValue);
			}
		},
	},
	computed: {
		hasCustomRenderer() {
			return hasCustomRenderer(this);
		},
	},
};

SingleList.generateQueryOptions = props => {
	const queryOptions = getQueryOptions(props);
	return getAggsQuery(queryOptions, props);
};
SingleList.defaultQuery = (value, props) => {
	let query = null;
	if (props.selectAllLabel && props.selectAllLabel === value) {
		if (props.showMissing) {
			query = { match_all: {} };
		}
		query = {
			exists: {
				field: props.dataField,
			},
		};
	}
	if (value) {
		if (props.showMissing && props.missingLabel === value) {
			query = {
				bool: {
					must_not: {
						exists: { field: props.dataField },
					},
				},
			};
		}
		query = {
			term: {
				[props.dataField]: value,
			},
		};
	}

	if (query && props.nestedField) {
		return {
			query: {
				nested: {
					path: props.nestedField,
					query,
				},
			},
		};
	}

	return query;
};
const mapStateToProps = (state, props) => ({
	options:
		props.nestedField && state.aggregations[props.componentId]
			? state.aggregations[props.componentId].reactivesearch_nested
			: state.aggregations[props.componentId],
	rawData: state.rawData[props.componentId],
	isLoading: state.isLoading[props.componentId],
	selectedValue:
		(state.selectedValues[props.componentId]
			&& state.selectedValues[props.componentId].value)
		|| '',
	themePreset: state.config.themePreset,
	error: state.error[props.componentId],
	componentProps: state.props[props.componentId],
});

const mapDispatchtoProps = {
	addComponent,
	removeComponent,
	setQueryOptions,
	setQueryListener,
	updateQuery,
	watchComponent,
	updateComponentProps,
	setComponentProps,
	setCustomQuery,
	setDefaultQuery,
};

const ListConnected = connect(mapStateToProps, mapDispatchtoProps)(SingleList);

SingleList.install = function(Vue) {
	Vue.component(SingleList.name, ListConnected);
};
export default SingleList;
