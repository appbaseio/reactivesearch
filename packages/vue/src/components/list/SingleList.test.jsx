import { mount } from '@vue/test-utils';
import { ListConnected as SingleList } from './SingleList.jsx';
import ReactiveBase from '../ReactiveBase/index.jsx';

const MOCK_AGGREGATIONS_DATA = {
	'authors.keyword': {
		buckets: [
			{
				key: 'J. K. Rowling',
				doc_count: 10,
			},
			{
				key: 'Nora Roberts',
				doc_count: 7,
			},
		],
	},
};

describe('SingleList', () => {
	it('should render no results message', () => {
		const wrapper = mount({
			name: 'TestComponent',
			render() {
				return (
					<ReactiveBase app="test" url="https://foo:bar@localhost:800">
						<SingleList
							mode="test"
							componentId="authors"
							dataField="authors.keyword"
							renderNoResults={() => 'No authors found'}
						/>
					</ReactiveBase>
				);
			},
		});

		expect(wrapper.element).toMatchSnapshot();
	});

	it('should render list of items', () => {
		const wrapper = mount({
			name: 'TestComponent',
			render() {
				return (
					<ReactiveBase app="test" url="https://foo:bar@localhost:800">
						<SingleList
							mode="test"
							componentId="authors"
							dataField="authors.keyword"
							renderNoResults={() => 'No authors found'}
							mockData={{
								aggregations: MOCK_AGGREGATIONS_DATA,
							}}
						/>
					</ReactiveBase>
				);
			},
		});

		expect(wrapper.element).toMatchSnapshot();
	});

	it('should render search/count/radio when showSearch/showCount/showRadio are true', () => {
		const wrapper = mount({
			name: 'TestComponent',
			render() {
				return (
					<ReactiveBase app="test" url="https://foo:bar@localhost:800">
						<SingleList
							mode="test"
							componentId="authors"
							dataField="authors.keyword"
							renderNoResults={() => 'No authors found'}
							mockData={{
								aggregations: MOCK_AGGREGATIONS_DATA,
							}}
							showSearch
							showCount
							showRadio
						/>
					</ReactiveBase>
				);
			},
		});

		expect(wrapper.element).toMatchSnapshot();
	});

	it('should not render search/count/radio when showSearch/showCount/showRadio are set to false', () => {
		const wrapper = mount({
			name: 'TestComponent',
			render() {
				return (
					<ReactiveBase app="test" url="https://foo:bar@localhost:800">
						<SingleList
							mode="test"
							componentId="authors"
							dataField="authors.keyword"
							renderNoResults={() => 'No authors found'}
							mockData={{
								aggregations: MOCK_AGGREGATIONS_DATA,
							}}
							showSearch={false}
							showCount={false}
							showRadio={false}
						/>
					</ReactiveBase>
				);
			},
		});

		expect(wrapper.element).toMatchSnapshot();
	});

	it('should use render prop to render the list', () => {
		const wrapper = mount({
			name: 'TestComponent',
			render() {
				return (
					<ReactiveBase app="test" url="https://foo:bar@localhost:800">
						<SingleList
							mode="test"
							componentId="authors"
							dataField="authors.keyword"
							renderNoResults={() => 'No authors found'}
							mockData={{
								aggregations: MOCK_AGGREGATIONS_DATA,
							}}
							render={({ data, handleChange }) => (
								<div>
									<ul>
										{/* eslint-disable camelcase */}
										{data.map(({ doc_count, key }) => (
											<li
												style={{ backgroundColor: 'transparent' }}
												key={key}
												onClick={() => handleChange(key)}
											>
												{key} --- {doc_count}
											</li>
										))}
									</ul>
								</div>
							)}
						/>
					</ReactiveBase>
				);
			},
		});

		expect(wrapper.element).toMatchSnapshot();
	});

	it('should use renderItem to render the list item', () => {
		const wrapper = mount({
			name: 'TestComponent',
			render() {
				return (
					<ReactiveBase app="test" url="https://foo:bar@localhost:800">
						<SingleList
							mode="test"
							componentId="authors"
							dataField="authors.keyword"
							renderNoResults={() => 'No authors found'}
							mockData={{
								aggregations: MOCK_AGGREGATIONS_DATA,
							}}
							renderItem={({ label, count }) => (
								<div>
									{label}
									<span style={{ marginLeft: 5, color: 'dodgerblue' }}>
										{count}
									</span>
								</div>
							)}
						/>
					</ReactiveBase>
				);
			},
		});

		expect(wrapper.element).toMatchSnapshot();
	});

	it('should select default value', () => {
		const wrapper = mount({
			name: 'TestComponent',
			render() {
				return (
					<ReactiveBase app="test" url="https://foo:bar@localhost:800">
						<SingleList
							mode="test"
							componentId="authors"
							dataField="authors.keyword"
							renderNoResults={() => 'No authors found'}
							mockData={{
								aggregations: MOCK_AGGREGATIONS_DATA,
							}}
							defaultValue="Nora Roberts"
						/>
					</ReactiveBase>
				);
			},
		});

		expect(wrapper.element).toMatchSnapshot();
	});
});
