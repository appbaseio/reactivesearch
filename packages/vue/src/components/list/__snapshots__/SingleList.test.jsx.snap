// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SingleList should not render search/count/radio when showSearch/showCount/showRadio are set to false 1`] = `
<div
  class="css-nw180v"
>
  <div
    class="css-m1gst5"
    mockdata="[object Object]"
    mode="test"
  >
    <ul
      class="css-bbs1vb"
    >
      <li
        class=""
      >
        <input
          class="css-1ug7eyp"
          id="authors-J. K. Rowling"
          name="authors"
          readonly="true"
          type="radio"
          value="J. K. Rowling"
        />
        <label
          for="authors-J. K. Rowling"
        >
          <span>
            J. K. Rowling
          </span>
        </label>
      </li>
      <li
        class=""
      >
        <input
          class="css-1ug7eyp"
          id="authors-Nora Roberts"
          name="authors"
          readonly="true"
          type="radio"
          value="Nora Roberts"
        />
        <label
          for="authors-Nora Roberts"
        >
          <span>
            Nora Roberts
          </span>
        </label>
      </li>
    </ul>
  </div>
</div>
`;

exports[`SingleList should render list of items 1`] = `
<div
  class="css-nw180v"
>
  <div
    class="css-m1gst5"
    mockdata="[object Object]"
    mode="test"
  >
    <input
      class="css-1n1dm2c"
      placeholder="Search"
      style="margin: 0px 0px 8px;"
      value=""
    />
    <ul
      class="css-bbs1vb"
    >
      <li
        class=""
      >
        <input
          class="css-ydwgt"
          id="authors-J. K. Rowling"
          name="authors"
          readonly="true"
          show="true"
          type="radio"
          value="J. K. Rowling"
        />
        <label
          for="authors-J. K. Rowling"
        >
          <span>
            J. K. Rowling
            <span>
               (10)
            </span>
          </span>
        </label>
      </li>
      <li
        class=""
      >
        <input
          class="css-ydwgt"
          id="authors-Nora Roberts"
          name="authors"
          readonly="true"
          show="true"
          type="radio"
          value="Nora Roberts"
        />
        <label
          for="authors-Nora Roberts"
        >
          <span>
            Nora Roberts
            <span>
               (7)
            </span>
          </span>
        </label>
      </li>
    </ul>
  </div>
</div>
`;

exports[`SingleList should render no results message 1`] = `
<div
  class="css-nw180v"
>
  <div
    class="css-m1gst5"
    mode="test"
  >
    <input
      class="css-1n1dm2c"
      placeholder="Search"
      style="margin: 0px 0px 8px;"
      value=""
    />
    <ul
      class="css-bbs1vb"
    >
      <p>
        No authors found
      </p>
    </ul>
  </div>
</div>
`;

exports[`SingleList should render search/count/radio when showSearch/showCount/showRadio are true 1`] = `
<div
  class="css-nw180v"
>
  <div
    class="css-m1gst5"
    mockdata="[object Object]"
    mode="test"
  >
    <input
      class="css-1n1dm2c"
      placeholder="Search"
      style="margin: 0px 0px 8px;"
      value=""
    />
    <ul
      class="css-bbs1vb"
    >
      <li
        class=""
      >
        <input
          class="css-ydwgt"
          id="authors-J. K. Rowling"
          name="authors"
          readonly="true"
          show="true"
          type="radio"
          value="J. K. Rowling"
        />
        <label
          for="authors-J. K. Rowling"
        >
          <span>
            J. K. Rowling
            <span>
               (10)
            </span>
          </span>
        </label>
      </li>
      <li
        class=""
      >
        <input
          class="css-ydwgt"
          id="authors-Nora Roberts"
          name="authors"
          readonly="true"
          show="true"
          type="radio"
          value="Nora Roberts"
        />
        <label
          for="authors-Nora Roberts"
        >
          <span>
            Nora Roberts
            <span>
               (7)
            </span>
          </span>
        </label>
      </li>
    </ul>
  </div>
</div>
`;

exports[`SingleList should select default value 1`] = `
<div
  class="css-nw180v"
>
  <div
    class="css-m1gst5"
    mockdata="[object Object]"
    mode="test"
  >
    <input
      class="css-1n1dm2c"
      placeholder="Search"
      style="margin: 0px 0px 8px;"
      value=""
    />
    <ul
      class="css-bbs1vb"
    >
      <li
        class=""
      >
        <input
          class="css-ydwgt"
          id="authors-J. K. Rowling"
          name="authors"
          readonly="true"
          show="true"
          type="radio"
          value="J. K. Rowling"
        />
        <label
          for="authors-J. K. Rowling"
        >
          <span>
            J. K. Rowling
            <span>
               (10)
            </span>
          </span>
        </label>
      </li>
      <li
        class="active"
      >
        <input
          class="css-ydwgt"
          id="authors-Nora Roberts"
          name="authors"
          readonly="true"
          show="true"
          type="radio"
          value="Nora Roberts"
        />
        <label
          for="authors-Nora Roberts"
        >
          <span>
            Nora Roberts
            <span>
               (7)
            </span>
          </span>
        </label>
      </li>
    </ul>
  </div>
</div>
`;

exports[`SingleList should use render prop to render the list 1`] = `
<div
  class="css-nw180v"
>
  <div
    class="css-m1gst5"
    mockdata="[object Object]"
    mode="test"
  >
    <input
      class="css-1n1dm2c"
      placeholder="Search"
      style="margin: 0px 0px 8px;"
      value=""
    />
    <div>
      <ul>
        <li
          style="background-color: transparent;"
        >
          J. K. Rowling --- 10
        </li>
        <li
          style="background-color: transparent;"
        >
          Nora Roberts --- 7
        </li>
      </ul>
    </div>
  </div>
</div>
`;

exports[`SingleList should use renderItem to render the list item 1`] = `
<div
  class="css-nw180v"
>
  <div
    class="css-m1gst5"
    mockdata="[object Object]"
    mode="test"
  >
    <input
      class="css-1n1dm2c"
      placeholder="Search"
      style="margin: 0px 0px 8px;"
      value=""
    />
    <ul
      class="css-bbs1vb"
    >
      <li
        class=""
      >
        <input
          class="css-ydwgt"
          id="authors-J. K. Rowling"
          name="authors"
          readonly="true"
          show="true"
          type="radio"
          value="J. K. Rowling"
        />
        <label
          for="authors-J. K. Rowling"
        >
          <div>
            J. K. Rowling
            <span
              style="color: dodgerblue;"
            >
              10
            </span>
          </div>
        </label>
      </li>
      <li
        class=""
      >
        <input
          class="css-ydwgt"
          id="authors-Nora Roberts"
          name="authors"
          readonly="true"
          show="true"
          type="radio"
          value="Nora Roberts"
        />
        <label
          for="authors-Nora Roberts"
        >
          <div>
            Nora Roberts
            <span
              style="color: dodgerblue;"
            >
              7
            </span>
          </div>
        </label>
      </li>
    </ul>
  </div>
</div>
`;
